### ISSUE TITLE
Require release notes and migration guides in the Bevy repo as part of PRs

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#16629]
I also like the idea of frontloading this work. Currently I'm liking the idea of putting release note markdown somewhere in the Bevy repo and requiring it as part of the review process. The trick is that we can't / shouldn't enforce this everywhere, so how do we build the process in such a way that (1) we don't encumber non-feature changes / changes that don't need release notes and (2) we don't accidentally _drop_ features by not forcing them through the process.

_Originally posted by @cart in https://github.com/bevyengine/bevy/discussions/16431#discussioncomment-11442803_

**Comment by alice-i-cecile** [comment]
> There's a CI action that checks that PRs with the label M-Needs-Migration-Guide have the proper section in their description
> 
> In the same way, we could check on a label if the release note markdown file is present on label M-Needs-Release-Note
> 
> I think it's OK to require the file to be present, but a PR shouldn't be blocked on its content being final
> 
- [@mockersf](https://github.com/bevyengine/bevy/discussions/16431#discussioncomment-11442862).

I agree with this approach! Rather than relying on the creation of a new file, we should instead validate if a file in the appropriate folder was changed at all, to allow for flexibility in the cases where we need to update / merge into an existing release note / migration guide.

We'll need to coordinate with the `bevy-website` repo here, and move the contents of https://github.com/bevyengine/bevy-website/tree/main/release-content into the Bevy repo (at least going forward). Then, as we prepare the 0.16 notes, we need to change the release note / migration guide website building to use those folders as the source of truth somehow. That point was discussed more in https://github.com/bevyengine/bevy/discussions/16430.

There's other valuable work to do here, like making sure short codes work and sample code can be compiled, but that should be saved for follow-up. To start, I just want a structure for people to write notes in the bevy repo.

**Comment by mockersf** [comment]
> Rather than relying on the creation of a new file, we should instead validate if a file in the appropriate folder was changed at all, to allow for flexibility in the cases where we need to update / merge into an existing release note / migration guide.

We can check that the PR number was mentioned in a file in that folder

**Comment by mockersf** [comment]
We'll also have to decide how to handle this folder content during RC: do we empty it once we created the release branch? or when we finally release the new version? what happens to PR needing a release note merged between those two dates?

**Comment by alice-i-cecile** [comment]
My preference is currently to leave it in the Bevy repo permanently, and then build the site from the repo like we do for `bevy_assets`. Failing that though, I guess we should only remove them once the release is cut, and copy them over after each release candidate??

**Comment by alice-i-cecile** [comment]
Effectively complete. #18441 covers the remaining work.
